api/tests/mocha/services/connected-user-log.spec.js

```diff
@@ -0,0 +1,69 @@
+const chai = require('chai');
+const sinon = require('sinon');
+const connectedUserLogService = require('../../../src/services/connected-user-log');
+const db = require('../../../src/db');
+
+let clock;
+
+describe('Connected Users Log service', () => {
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    clock.restore();
+  });
+
+  describe('saveLog()', () => {
+    it('should save a log', () => {
+      sinon.stub(db.medicLogs, 'get').rejects({ status: 404 });
+      const putStub = sinon.stub(db.medicLogs, 'put').resolves();
+      const expectedDoc = {
+        _id: 'connected-user-userXYZ',
+        user: 'userXYZ',
+        timestamp: 0
+      };
+
+      return connectedUserLogService
+        .save('userXYZ')
+        .then(() => {
+          chai.expect(putStub.args[0][0]).to.deep.include(expectedDoc);
+        });
+    });
+
+    it('should throw any error trhat is not a 404', () => {
+      sinon.stub(db.medicLogs, 'get').rejects({ status: 401 });
+
+      return connectedUserLogService
+        .save('userXYZ')
+        .catch(err => {
+          chai.expect(err).to.deep.equal({ status: 401 });
+        });
+    });
+
+    it('should not save the log if the timestamp difference is less than the set interval', () => {
+      sinon.stub(db.medicLogs, 'get').resolves({ timestamp: 1 });
+      const putStub = sinon.stub(db.medicLogs, 'put');
+      clock.tick(25 * 60 * 1000); //25 minutes
+
+      return connectedUserLogService
+        .save('userXYZ')
+        .then(() => {
+          chai.expect(putStub.callCount).to.equal(0);
+        });
+    });
+
+    it('should save the log if the timestamp difference is more than the set interval', () => {
+      sinon.stub(db.medicLogs, 'get').resolves({ timestamp: 1 });
+      const putStub = sinon.stub(db.medicLogs, 'put');
+      clock.tick(31 * 60 * 1000); //31 minutes
```

Comment:
I don't think you need to move the clock because you've mocked the `get` to return a time a long time ago. Try dropping this line. Or if you think it's a better test, you could change the mock to return the current timestamp, and keep the clock tick.

Reply:
The clock is reset with each test so this [line that returns the current time](https://github.com/medic/cht-core/pull/7116/files#diff-bd42b9ba0b8861671a153b3c67acec204ff871d34095170109aecfcdbd929ba5R23) would return 0 without the tick.

Reply:
Right. I guess I'm thinking you don't need to mock the clock and reset it at all, unless another test requires it?

Reply:
I use it [here](https://github.com/medic/cht-core/pull/7116/files/77bbb36cccd9ab5a598111f0c3d64427edd2b44d#diff-8c37c72e4d1e41bef4c72b2522cbc1c4066d8f1f608e05ee7f9e0ac103625616R22) to predictably assert the full object. Leaving it as it is.